### DISCUSSION TITLE
Add the ability to search metadata and title with single prefix

### DISCUSSION THREAD

**Discussion by PenDrag0n7** [gotson/komga#1569] [Ideas]
#### Feature
I would like the ability to be able to search title and all meta data that contains a search term, without having to specify individual  prefix. 

The goal is to have have a prefix to make searching easy convenient, has multiple uses and overall simpler/ less "wordy". 
Especially when tying longer and more complex search on mobile and Tachyomi.

Expressions (A) Short hand
Expression (B) Full form

#### Examples
1. Search any containing Horror
This should search tags, genre, author, title etc to find if the term "horror" exist in any of them and return all matching results.
Expression (A) => +"Horror"
Expression (B) => any:"Horror"

2. Search any containing Horror, Action, Manga (Implicit OR)
When given multiple search terms this should be 
Expression (A) => +"Horror"
Expression (B) => any:"Horror"

2. Search any not containing 
This should return results that contain "Horror" but does NOT contain "Manga"
Expressions (A) -> +"Horror" -"Manga"
Expressions (B) -> any:Horror NOT any:Manga

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by gotson** [comment]
> Expressions (A) Short hand

this is not possible, by default some fields are searched depending on the entity type, as described [here](https://komga.org/guides/search.html):

> By default, the search will match the item title. For Books, the ISBN will also be matched by default.

> Especially when tying longer and more complex search on mobile and Tachyomi.

Can you clarify the use case about Tachiyomi ?

**Comment by PenDrag0n7** [comment]
With Tachiyomi when you select a Manga then click one of the tags, the app just enters the word in search without the appropriate prefix which by komga's default behaviour only searches the title. 

giving it a default prefix "any:" will help give better results, unless this is a know bug  and there is already something in the works.

**Comment by gotson** [comment]
> With Tachiyomi when you select a Manga then click one of the tags, the app just enters the word in search without the appropriate prefix which by komga's default behaviour only searches the title.

That's a Tachiyomi limitations, it's doing a global search, not knowing how to perform a tag search depending on each extension.

Having an `any` field would not address this case, as Tachiyomi would not change the behaviour, since it's global and not specific to the Komga extension.
